packages/calcite-components/src/components/input-number/input-number.tsx

```diff
@@ -840,12 +841,19 @@ export class InputNumber
       useGrouping: this.groupSeparator
     };
 
-    const sanitizedValue = sanitizeNumberString(
-      // no need to delocalize a string that ia already in latn numerals
-      (this.numberingSystem && this.numberingSystem !== "latn") || defaultNumberingSystem !== "latn"
-        ? numberStringFormatter.delocalize(value)
-        : value
-    );
+    const isValueDeleted =
+      this.previousValue?.length > value.length || this.value?.length > value.length;
+    const hasTrailingDecimalSeparator = value.charAt(value.length - 1) === ".";
+    const sanitizedValue =
+      hasTrailingDecimalSeparator && isValueDeleted
+        ? value
```

Comment:
Is there a reason why the value doesn't need to be delocalized when there is a trailing 0 and it is a delete action?

Reply:
i assume you are referring to `trailingDecimalSeparator` , the value is already delocalized by then.

Reply:
If the value is already delocalized by then, we could remove the delocalization a couple lines down right?

Reply:
i think so, it was added [here](https://github.com/Esri/calcite-components/pull/5427/files#diff-fe3d603d6390f87fdcdd3f40e34778ccdb07ff9edc52e440ad13c9571cfceadaR841) . When removed , all the tests are passing which is a good sign.

Reply:
Are we sure it is already delocalized in all cases? (e.g. typing, pasting, and setting the value via JS?)?

Reply:
yup, while typing or pasting InputNumberInput handler will delocalize the strings. For setting the value via JS, i don't think users can parse non latin number values which sounds strange tbh. I guess it is intentional to avoid localization twice.